firmware/projects/Demo/CAN/Send/main.cc

```diff
@@ -4,29 +4,22 @@
 #include <cstdint>
 
 #include "bindings.h"
-#include "generated/can/demobus_can_messages.h"
-#include "generated/can/demobus_msg_registry.h"
-#include "shared/comms/can/can_bus.h"
+#include "generated/can/demobus_bus.h"
+#include "generated/can/demobus_messages.h"
 
-generated::can::DemobusMsgRegistry veh_can_registry{};
+using namespace generated::can;
 
-shared::can::CanBus veh_can_bus{
-    bindings::veh_can_base,
-    veh_can_registry,
-};
+DemobusBus can_bus{bindings::demo_can_base};
 
 int main(void) {
     bindings::Initialize();
     uint32_t interval_ms = 50;
 
-    generated::can::ButtonStatus btn_msg{};
-
     while (true) {
-        veh_can_bus.Update();
-
-        btn_msg.state = bindings::button.Read();
-
-        veh_can_bus.Send(btn_msg);
+        TxButtonStatus msg{
+            .state = bindings::button.Read(),
+        };
+        can_bus.Send(msg.pack());
```

Comment:
Not a huge fan of having to call pack on the message when passing it into the Send function. Why not make it a template with a concept that makes sure the pack function exists?

Reply:
Sure then leave it out but still why not have the send method be templates and not require the user to pack the message

Reply:
Re-added concepts. All other convos are resolved so I'm going to merge